consensus/ibft/ibft.go

```diff
@@ -1287,12 +1286,32 @@ func (i *Ibft) GetBlockCreator(header *types.Header) (types.Address, error) {
 	return ecrecoverFromHeader(header)
 }
 
+// PreStateCommit a hook to be called before finalizing state transition on inserting block
+func (i *Ibft) runInsertTransactionsHook(
+	header *types.Header,
+	transition *state.Transition,
+	transactions *[]*types.Transaction,
```

Comment:
This can probably be a struct param

Reply:
@zivkovicmilos What do you mean?

Reply:
I'm just suggesting you bundle these three params into a struct - I actually think you already have it somewhere